src/org/traccar/events/GeofenceEventHandler.java

```diff
@@ -0,0 +1,74 @@
+package org.traccar.events;
+
+import java.sql.SQLException;
+import java.util.Set;
+
+import org.traccar.BaseEventHandler;
+import org.traccar.Context;
+import org.traccar.database.DataManager;
+import org.traccar.database.GeofenceManager;
+import org.traccar.helper.Log;
+import org.traccar.model.Device;
+import org.traccar.model.Event;
+import org.traccar.model.Position;
+
+public class GeofenceEventHandler extends BaseEventHandler {
+
+    private int suppressRepeated;
+    private GeofenceManager geofenceManager;
+    private DataManager dataManager;
+
+    public GeofenceEventHandler() {
+        suppressRepeated = Context.getConfig().getInteger("event.suppressRepeated", 60);
+        geofenceManager = Context.getGeofenceManager();
+        dataManager = Context.getDataManager();
+    }
+
+    @Override
+    protected Event analizePosition(Position position) {
+        Event event = null;
+        if (!isLastPosition() || !position.getValid()) {
+            return event;
+        }
+
+        Device device = dataManager.getDeviceById(position.getDeviceId());
+        if (device == null) {
+            return event;
+        }
+
+        Set<Long> geofences = geofenceManager.getAllDeviceGeofences(position.getDeviceId());
+        if (geofences == null) {
```

Comment:
I think it should control each geofence as separate in/out state, for doing this the BaseEventhandler should accept the return of an list of events instead of a single event. In this case will be possible to return more than one event like : 

EXIT : LOCATION 1
ENTER: LOCATION 2
ENTER: LOCATION 3

what you think about this ?